### PR TITLE
New project modal + ProcessModelType enum

### PR DESCRIPTION
**Pull request by askepen**
## Describe your changes
Add popup for creating projects

- [x] Select net type
- [x] Editable project name 
- [x] Editable description

## Checklist before requesting a review
- [x] I have performed a self-review of my code

### CODE REVIEW

**Review by MathiasBoi6 — APPROVED** [review]
Looks great

**Comment by askepen** [comment]
What should the behaviour be when creating a new project:
- An empty graph?
- An example graph depending on the graph type (DCR / Petri-net / flowchart / etc.)?

**Comment by leny-mi** [comment]
If I click on the plus button, then update the modal, close it and reopen it still has the previous changes. I'm unsure whether this is expected behaviour but wanted to mention it

File: packages/synthchron-simulator/src/types/processModelTypes/dcrGraphTypes.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import { Point } from '../processModel'
+import { Point, ProcessModelType } from '../processModel'
 
 export interface DcrGraphProcessModel {
-  type: 'dcr-graph'
+  type: ProcessModelType
```
[the remark this answers is not in the document]
Comment (leny-mi) [reply packages/synthchron-simulator/src/types/processModelTypes/dcrGraphTypes.ts:4]: Can we do something like the following?
```
export interface DcrGraphProcessModel {
  type: ProcessModelType.DcrGraph
  ...
}
```

[the remark this answers is not in the document]
Comment (askepen) [reply packages/synthchron-simulator/src/types/processModelTypes/dcrGraphTypes.ts:4]: Would this work?
```suggestion
  type: ProcessModelType.DcrGraph
```

[the remark this answers is not in the document]
Comment (leny-mi) [reply packages/synthchron-simulator/src/types/processModelTypes/dcrGraphTypes.ts:4]: I think it should :+1:

**Comment by askepen** [comment]
> If I click on the plus button, then update the modal, close it and reopen it still has the previous changes. I'm unsure whether this is expected behaviour but wanted to mention it

I made this intentionally, so if the user accidentally closes the modal their very important name and description is not lost 😅

**Review by leny-mi — APPROVED** [review]
looks great!

**Pull request merged**